src/snowflake/snowpark/session.py

```diff
@@ -823,19 +839,28 @@ def clear_packages(self) -> None:
         """
         self._packages.clear()
 
-    def add_requirements(self, file_path: str, force_push: bool = True) -> None:
+    def add_requirements(
+        self,
+        file_path: str,
+        *,
+        force_push: bool = False,
+    ) -> None:
         """
         Adds a `requirement file <https://pip.pypa.io/en/stable/user_guide/#requirements-files>`_
         that contains a list of packages as dependencies of a user-defined function (UDF). Pure Python packages that
         are not available in Snowflake will be pip installed locally and made available as an import (via zip file
         on a remote stage).
 
-         Note that this function also supports addition of requirements via a `conda environment yaml file
-         <https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually>_`.
+        Note that this function also supports addition of requirements via a `conda environment yaml file <https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually>_`.
+
+        EXPERIMENTAL: Upon setting `Session.custom_packages_upload_enabled` to True, Pure Python packages that are not available in
+        Snowflake will be pip installed locally and made available as an import (via zip file on a remote stage).
+        Note that if you wish to use a specific version of pip, you can set the environment variable `PIP_PATH` to your
+        pip executable. This feature is experimental and works well on UNIX systems only, do not use it in production!
 
         Args:
             file_path: The path of a local requirement file.
-            force_push: Force upload Python packages with native dependencies.
+            force_push: Force upload Python packages with native dependencies (experimental).
 
```

Comment:
nit: can you use the same description for `force_push` for all doc strings:

`force_push: Force upload unavailable Python packages which contain native C/C++ code (experimental).`